### ISSUE TITLE
[arithmetic_side_effects] Paths that refer literals should not trigger the lint

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#11262]
```rust
#![warn(clippy::arithmetic_side_effects)]

fn main() {
    let a = 1;
    let _b = 2 / a;
}
```

As seen above, `2 / 1` is  obviously a valid arithmetic operation.
